pkg/util/nodelock/nodelock.go

```diff
@@ -90,22 +103,27 @@ func ReleaseNodeLock(nodeName string, lockname string, pod *corev1.Pod, timeout
 			return nil
 		}
 	}
-	patchData := fmt.Sprintf(`{"metadata":{"annotations":{"%s":null},"resourceVersion":"%s"}}`, NodeLockKey, node.ResourceVersion)
-	_, err = client.GetClient().CoreV1().Nodes().Patch(ctx, nodeName, types.MergePatchType, []byte(patchData), metav1.PatchOptions{})
-	for i := 0; i < MaxLockRetry && err != nil; i++ {
-		klog.ErrorS(err, "Failed to update node", "node", nodeName, "retry", i)
-		time.Sleep(100 * time.Millisecond)
+	err = retry.OnError(DefaultStrategy, func(err error) bool {
+		// Retry on any error
+		return true
+	}, func() error {
 		node, err = client.GetClient().CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
 		if err != nil {
-			klog.ErrorS(err, "Failed to get node when retry to update", "node", nodeName)
-			continue
+			klog.ErrorS(err, "Failed to get node when retry to patch", "node", nodeName)
+			return err
 		}
 		patchData := fmt.Sprintf(`{"metadata":{"annotations":{"%s":null},"resourceVersion":"%s"}}`, NodeLockKey, node.ResourceVersion)
 		_, err = client.GetClient().CoreV1().Nodes().Patch(ctx, nodeName, types.MergePatchType, []byte(patchData), metav1.PatchOptions{})
-	}
+		if err != nil {
+			klog.ErrorS(err, "Failed to patch node when retry to patch", "node", nodeName)
+			return err
+		}
+		return nil
+	})
 	if err != nil {
-		return fmt.Errorf("releaseNodeLock exceeds retry count %d", MaxLockRetry)
+		return fmt.Errorf("failed to release node lock on node %s: %w", nodeName, err)
```

Comment:
@mayooot I think it is better to record the information of `DefaultStrategy` in the error message.